timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/CorruptionDetectionIntegrationTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.corruption.detection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.SetMultimap;
+import com.palantir.paxos.Client;
+import com.palantir.paxos.ImmutableNamespaceAndUseCase;
+import com.palantir.paxos.NamespaceAndUseCase;
+import java.util.List;
+import java.util.stream.IntStream;
+import org.junit.Test;
+
+/**
+ * This class performs integration tests by inducing and detecting corruption in one or more series.
+ * Note - All tests only induce and detect one type of corruption check violation -> ACCEPTED_VALUE_GREATER_THAN_LEARNED
+ */
+public class CorruptionDetectionIntegrationTest extends TimeLockCorruptionTestSetup {
+
+    @Test
+    public void detectCorruptionForLogAtSeqAtBatchEnd() {
+        // We write logs in range - [1, 500]. The first range of sequences for corruption detection = [0, 499] since
+        // this range is computed from INITIAL_PROGRESS = -1.
+        writeLogsOnLocalAndRemote(1, 500);
+        induceGreaterAcceptedValueCorruption(499);
+        assertDetectedViolations(ImmutableSet.of(CorruptionCheckViolation.ACCEPTED_VALUE_GREATER_THAN_LEARNED));
```

Comment:
minor nit: `assertViolationsDetected`

Reply:
also, split into two methods: some violations (and maybe can be specific), or no violation.